<div style="
    display: flex;
    justify-content: center;
    align-items: center;
    width: 70%;
    margin: 40px auto;
    border-radius: 12px;
    background: linear-gradient(45deg, #ff6a00, #ffcc00, #ff00aa, #00fff2, #ff6a00);
    background-size: 300% 300%;
    animation: borderFlow 6s linear infinite;
    padding: 2px; /* bordure fine autour du orange */
    box-sizing: border-box;
">
    <!-- Bloc intérieur -->
    <div style="
        display: flex;
        align-items: center;
        justify-content: center;
        flex-wrap: wrap;
        background: linear-gradient(90deg, #ff7b00ff, #ff9500);
        color: white;
        font-family: 'Segoe UI', Arial, sans-serif;
        font-size: 28px;
        font-weight: 600;
        letter-spacing: 2px;
        padding: 20px 40px;
        border-radius: 10px; /* légèrement inférieur au parent */
        text-transform: uppercase;
        box-shadow: inset 0 0 0 0 rgba(0,0,0,0), 0 3px 8px rgba(0,0,0,0.25);
        width: 100%;
        height: 100%;
        box-sizing: border-box;
    ">
        <img src="assets/img/bottleneck.png" 
             alt="Logo Bottleneck" 
             style="height: 120px; width: auto; filter: drop-shadow(0 0 6px rgba(0,0,0,0.3)); margin-right: 25px;">
        <div style="text-align: center;">
            Analyse du stock et des ventes du site<br>
            Bottleneck
        </div>
    </div>
</div>

<style>
@keyframes borderFlow {
  0% { background-position: 0% 50%; }
  50% { background-position: 100% 50%; }
  100% { background-position: 0% 50%; }
}
</style>

# 🎯 Objectif du projet — Bottleneck 🍷

Ce notebook a pour objectif de **centraliser, nettoyer et analyser les données de ventes et de stocks de l’entreprise Bottleneck**, afin d’en extraire des **insights pertinents pour le comité de direction (CODIR)**.

---

## 🧩 Contexte

L’entreprise Bottleneck utilise actuellement plusieurs sources de données indépendantes :
- Une **extraction de l’ERP** (références produits, prix, état du stock),
- Une **extraction du site Web** (SKU, quantités vendues, descriptions produits, etc.),
- Une **table de liaison** permettant de relier les références entre les deux systèmes.

Cependant, des divergences existent entre ces fichiers (erreurs de saisie, types de données incohérents, références non correspondantes), rendant les analyses complexes.

---

## 📈 Objectifs principaux

Le travail se décompose en deux phases :

### Phase 1 — Agrégation et fiabilisation des données
- Rapprocher les extractions **ERP** et **Web** via la table de liaison,  
- Identifier les **erreurs et incohérences** dans les données (saisie, calcul, jointure, etc.),  
- Proposer des **solutions d’amélioration** pour fiabiliser les données en amont.

### Phase 2 — Analyses statistiques et interprétation
- Calcul du **chiffre d’affaires total et par produit**,  
- Détection et visualisation des **valeurs aberrantes** (Z-Score, écart interquartile, boxplot),  
- Étude du **taux de marge**, de la **rotation des stocks** et des **mois de couverture**,  
- Analyse des **corrélations** entre les variables quantitatives : prix, prix d’achat, stock, ventes, marge, etc.

---

## 🧠 Finalité

Les résultats présentés dans ce notebook serviront :
- À alimenter la **prise de décision du CODIR**,  
- À préparer la **future refonte du système de data visualisation**,  
- Et à définir une **base de données conforme et cohérente**, respectant les exigences **RGPD**.

---

> 💡 *L’ensemble de cette étude a été menée dans une optique de clarté, de rigueur et de reproductibilité, afin d’offrir à Bottleneck une vision fiable de ses performances commerciales.*


<a id="top"></a>

<h2 style="color:RGB(51,165,182);">📘 Sommaire</h2>

<ul style="list-style:none; line-height:1.8; font-size:17px; padding-left:0;">
  <li>
    <a href="#etape1" style="color:RGB(51,165,182); text-decoration:none;">
      <img src="assets/img/house-solid-full.svg" width="15" style="vertical-align:middle; margin-right:6px; filter: invert(64%) sepia(53%) saturate(379%) hue-rotate(150deg) brightness(95%) contrast(92%);">
      Étape 1 - Importation des librairies et chargement des fichiers
    </a>
    <ul style="list-style:none; margin-left:20px;">
      <li><a href="#etape1_1" style="color:RGB(51,165,182); text-decoration:none;">📘 1.1 - Importation des librairies</a></li>
      <li><a href="#etape1_2" style="color:RGB(51,165,182); text-decoration:none;">📗 1.2 - Chargement des fichiers</a></li>
    </ul>
  </li>

  <li>
    <a href="#etape2" style="color:RGB(51,165,182); text-decoration:none;">
      <img src="assets/img/house-solid-full.svg" width="15" style="vertical-align:middle; margin-right:6px; filter: invert(64%) sepia(53%) saturate(379%) hue-rotate(150deg) brightness(95%) contrast(92%);">
      Étape 2 - Analyse exploratoire des fichiers
    </a>
    <ul style="list-style:none; margin-left:20px;">
      <li><a href="#etape1_2" style="color:RGB(51,165,182); text-decoration:none;">📊 2.1 - Analyse exploratoire du fichier erp.xlsx</a></li>
      <li><a href="#etape1_1" style="color:RGB(51,165,182); text-decoration:none;">📊 2.1.1 - Analyse exploratoire de chaque variable du fichier erp.xlsx</a></li>
      <ul style="margin-left:20px; list-style:none;">
        <li><a href="#etape2_1_1" style="color:RGB(51,165,182); text-decoration:none;">🔹 2.1.1.1 - Analyse de la variable PRIX</a></li>
        <li><a href="#etape2_1_1_2" style="color:RGB(51,165,182); text-decoration:none;">🔹 2.1.1.2 - Analyse de la variable STOCK</a></li>
        <li><a href="#etape2_1_1_3" style="color:RGB(51,165,182); text-decoration:none;">🔹 2.1.1.3 - Analyse de la variable ONSALE_WEB</a></li>
        <li><a href="#etape2_1_1_4" style="color:RGB(51,165,182); text-decoration:none;">🔹 2.1.1.4 - Analyse de la variable prix d'achat</a></li>
      </ul>
      <li><a href="#etape2_2" style="color:RGB(51,165,182); text-decoration:none;">📊 2.2 - Analyse exploratoire du fichier web.xlsx</a></li>
      <li><a href="#etape2_3" style="color:RGB(51,165,182); text-decoration:none;">📊 2.3 - Analyse exploratoire du fichier liaison.xlsx</a></li>
    </ul>
  </li>

  <li>
    <a href="#etape3" style="color:RGB(51,165,182); text-decoration:none;">
      <img src="assets/img/house-solid-full.svg" width="15" style="vertical-align:middle; margin-right:6px; filter: invert(64%) sepia(53%) saturate(379%) hue-rotate(150deg) brightness(95%) contrast(92%);">
      Étape 3 - Jonction des fichiers
    </a>
    <ul style="list-style:none; margin-left:20px;">
      <li><a href="#etape3_1" style="color:RGB(51,165,182); text-decoration:none;">🔸 3.1 - Jonction du fichier df_erp et df_liaison</a></li>
      <li><a href="#etape3_2" style="color:RGB(51,165,182); text-decoration:none;">🔸 3.2 - Jonction du fichier df_merge et df_web</a></li>
    </ul>
  </li>

  <li>
    <a href="#etape4" style="color:RGB(51,165,182); text-decoration:none;">
      <img src="assets/img/house-solid-full.svg" width="15" style="vertical-align:middle; margin-right:6px; filter: invert(64%) sepia(53%) saturate(379%) hue-rotate(150deg) brightness(95%) contrast(92%);">
      Étape 4 - Analyse univariée des prix
    </a>
    <ul style="list-style:none; margin-left:20px;">
      <li><a href="#etape4_1" style="color:RGB(51,165,182); text-decoration:none;">📈 4.1 - Exploration par la visualisation de données</a></li>
      <li><a href="#etape4_2" style="color:RGB(51,165,182); text-decoration:none;">📉 4.2 - Exploration par l'utilisation de méthodes statistiques</a></li>
      <ul style="margin-left:20px; list-style:none;">
        <li><a href="#etape4_2_1" style="color:RGB(51,165,182); text-decoration:none;">🔹 4.2.1 - Identification par le Z-index</a></li>
        <li><a href="#etape4_2_2" style="color:RGB(51,165,182); text-decoration:none;">🔹 4.2.2 - Identification par l'intervalle interquartile</a></li>
      </ul>
    </ul>
  </li>

  <li>
    <a href="#etape5" style="color:RGB(51,165,182); text-decoration:none;">
      <img src="assets/img/house-solid-full.svg" width="15" style="vertical-align:middle; margin-right:6px; filter: invert(64%) sepia(53%) saturate(379%) hue-rotate(150deg) brightness(95%) contrast(92%);">
      Étape 5 - Analyse univariée du CA, des quantités vendues, des stocks et de la marge ainsi qu'une analyse multivariée
    </a>
    <ul style="list-style:none; margin-left:20px;">
      <li><a href="#etape5_1" style="color:RGB(51,165,182); text-decoration:none;">💰 5.1 - Analyse des ventes en CA</a></li>
      <li><a href="#etape5_2" style="color:RGB(51,165,182); text-decoration:none;">📦 5.2 - Analyse des ventes en quantité</a></li>
      <li><a href="#etape5_3" style="color:RGB(51,165,182); text-decoration:none;">📊 5.3 - Analyse des stocks</a></li>
      <li><a href="#etape5_4" style="color:RGB(51,165,182); text-decoration:none;">📈 5.4 - Analyse du taux de marge</a></li>
      <li><a href="#etape5_5" style="color:RGB(51,165,182); text-decoration:none;">🔗 5.5 - Corrélations entre stock, sales et price</a></li>
      <li><a href="#etape5_6" style="color:RGB(51,165,182); text-decoration:none;">🧾 5.6 - Mise à disposition du fichier Excel final</a></li>
    </ul>
  </li>
</ul>


<a id="etape1"></a>
<div style="
  border: 1px solid RGB(0,150,250);
  border-radius:5px;
  background-color:rgba(125, 194, 204, 0.15);
  margin:auto;
">

  <h2 style="
    margin:auto;
    padding:15px 20px;
    background-color:RGB(51,165,182);
    color:#fff;
    display:flex;
    align-items:center;
    border-radius:5px;
  ">
    <a href="#top" style="text-decoration:none; display:flex; align-items:center; margin-right:10px;">
      <img src="assets/img/house-solid-full.svg" width="30" style="vertical-align:middle;">
    </a>
    Étape 1 - Importation des librairies et chargement des fichiers
  </h2>
</div>

<a id="etape1_1"></a>
<div style="border: 1px solid RGB(0,150,250); border-radius:5px; background-color:rgba(51,165,182,0.05);">

  <h3 style="margin:auto; padding:15px 20px; color:RGB(51,165,182); display:flex; align-items:center;">
    <a href="#top" style="text-decoration:none; display:flex; align-items:center; margin-right:10px;">
      <img src="assets/img/house-solid-full.svg" width="16"
           style="vertical-align:middle; filter: invert(64%) sepia(53%) saturate(379%) hue-rotate(150deg) brightness(95%) contrast(92%);">
    </a>
    1.1 - Importation des librairies
  </h3>
</div>


In [270]:
#Importation de la librairie Pandas
from IPython.display import display, HTML
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
warnings.filterwarnings("ignore", category=UserWarning, module="openpyxl")

print("Tout est installé et prêt 🎉")

Tout est installé et prêt 🎉


In [271]:
#Importation de la librairie plotly express (attention peut bug)
import plotly.express as px

In [ ]:
# Affichage de toutes les colonnes du dataframe
pd.set_option('display.max_columns', None)


<div style="background-color:#1e1e1e; color:#dcdcdc; padding:10px 15px; border-radius:8px; font-family:Consolas, monospace; border:1px solid #444; width:fit-content; margin:auto;">

<b>Réponse :</b><br>
<span style="color:#9CDCFE;">pd</span>.<span style="color:#DCDCAA;">set_option</span>(
<span style="color:#CE9178;">'display.max_columns'</span>, 
<span style="color:#569CD6;">None</span>)
<div style="color:#32CD32; font-weight:bold; font-family:Segoe UI, sans-serif; font-size:15px; text-align:center; margin-top:10px;">
✅ Toutes les colonnes du DataFrame seront désormais affichées.
</div>
</div>

<a id="etape1_2"></a>
<div style="border: 1px solid RGB(0,150,250); border-radius:5px; background-color:rgba(51,165,182,0.05);">

  <h3 style="margin:auto; padding:15px 20px; color:RGB(51,165,182); display:flex; align-items:center;">
    <a href="#top" style="text-decoration:none; display:flex; align-items:center; margin-right:10px;">
      <img src="assets/img/house-solid-full.svg" width="16"
           style="vertical-align:middle; filter: invert(64%) sepia(53%) saturate(379%) hue-rotate(150deg) brightness(95%) contrast(92%);">
    </a>
    1.2 - Chargements des fichiers
  </h3>
</div>

In [273]:
#Importation du fichier web.xlsx
df_web = pd.read_excel("data/web.xlsx")

#Importation du fichier erp.xlsx
df_erp = pd.read_excel("data/erp.xlsx")

#Importation du fichier liaison.xlsx
df_liason = pd.read_excel("data/liaison.xlsx")


<div style="
    background: #4CAF50;
    color:black;
    padding:15px 40px;
    border-radius:10px;
    font-family:'Segoe UI', Arial, sans-serif;
    font-size:16px;
    text-align:center;
    box-shadow:0 4px 12px rgba(0,0,0,0.3);
    width:fit-content;
    margin:30px auto;
">
<b>Résultat :</b> 🔥 Importation terminée :<br>
 les fichiers <b>web.xlsx</b>, <b>erp.xlsx</b> et <b>liaison.xlsx</b> sont chargés avec succès !
</div>

<a id="etape2"></a>
<div style="
  border: 1px solid RGB(0,150,250);
  border-radius:5px;
  background-color:rgba(125, 194, 204, 0.15);
  margin:auto;
">

  <h2 style="
    margin:auto;
    padding:15px 20px;
    background-color:RGB(51,165,182);
    color:#fff;
    display:flex;
    align-items:center;
    border-radius:5px;
  ">
    <a href="#top" style="text-decoration:none; display:flex; align-items:center; margin-right:10px;">
      <img src="assets/img/house-solid-full.svg" width="30" style="vertical-align:middle;">
    </a>
    Etape 2 - Analyse exploratoire des fichiers
  </h2>
</div>

<a id="etape1_2"></a>
<div style="border: 1px solid RGB(0,150,250); border-radius:5px; background-color:rgba(51,165,182,0.05);">

  <h3 style="margin:auto; padding:15px 20px; color:RGB(51,165,182); display:flex; align-items:center;">
    <a href="#top" style="text-decoration:none; display:flex; align-items:center; margin-right:10px;">
      <img src="assets/img/house-solid-full.svg" width="16"
           style="vertical-align:middle; filter: invert(64%) sepia(53%) saturate(379%) hue-rotate(150deg) brightness(95%) contrast(92%);">
    </a>
    2.1 - Analyse exploratoire du fichier erp.xlsx
  </h3>
</div>

In [274]:
#Afficher les dimensions du dataset
rows, cols = df_erp.shape
display(HTML(f"""
<p style='font-family: Segoe UI, Arial; font-size:18px; color:#e0e0e0;'>
Le tableau comporte <span style='color:#ffcc00; font-weight:bold;'>{rows:,}</span> observation(s) ou article(s)<br>
Le tableau comporte <span style='color:#ffcc00; font-weight:bold;'>{cols}</span> colonne(s)
</p>
"""))

In [275]:
#Consulter le nombre de colonnes
print("Nombre de colonnes :", df_erp.shape[1])

#La nature des données dans chacune des colonnes
print("\n\033[96mTypes de données :\033[0m")
print(df_erp.dtypes)

#Le nombre de valeurs présentes dans chacune des colonnes
print("\n\033[96mValeurs non nulles par colonne :\033[0m")
print(df_erp.count())


Nombre de colonnes : 6

Types de données :
product_id          int64
onsale_web          int64
price             float64
stock_quantity      int64
stock_status       object
purchase_price    float64
dtype: object

Valeurs non nulles par colonne :
product_id        825
onsale_web        825
price             825
stock_quantity    825
stock_status      825
purchase_price    825
dtype: int64


In [276]:
#Afficher les 5 premières lignes de la table
df_erp.head()

product_id  onsale_web  price  stock_quantity stock_status  purchase_price
0        3847           1   24.2              16      instock           12.88
1        3849           1   34.3              10      instock           17.54
2        3850           1   20.8               0   outofstock           10.64
3        4032           1   14.1              26      instock            6.92
4        4039           1   46.0               3   outofstock           23.77

In [277]:
#Vérifier si il y a des lignes en doublon dans la colonne product_id
df_erp["product_id"].duplicated().sum()

np.int64(0)

In [278]:
#Afficher les valeurs distinctes de la colonne stock_status
print("\033[96mValeurs distinctes de stock_status :\033[0m", df_erp["stock_status"].unique())

# Fréquence d'apparition
print("\n\033[96mFréquence de chaque valeur :\033[0m")
print(df_erp["stock_status"].value_counts())

# Relation entre stock_status et stock_quantity
print("\n\033[96mCorrélation entre 'stock_status' et 'stock_quantity' :\033[0m")
pd.crosstab(df_erp["stock_status"], df_erp["stock_quantity"])

Valeurs distinctes de stock_status : ['instock' 'outofstock']

Fréquence de chaque valeur :
stock_status
instock       733
outofstock     92
Name: count, dtype: int64

Corrélation entre 'stock_status' et 'stock_quantity' :


stock_quantity  -10   -1     0     1     2     3     4     5     6     7    \
stock_status                                                                 
instock            0     0     1    12     6    17    21    27    21    32   
outofstock         1     1    89     0     0     1     0     0     0     0   

stock_quantity   8     9     10    11    12    13    14    15    16    17   \
stock_status                                                                 
instock           21    17    19    14    29    10    18    19    21    13   
outofstock         0     0     0     0     0     0     0     0     0     0   

stock_quantity   18    19    20    21    22    23    24    25    26    27   \
stock_status                                                                 
instock           16    14    10    14    13    30    21    12    18    16   
outofstock         0     0     0     0     0     0     0     0     0     0   

stock_quantity   28    29    30    31    32    33    34    35    36    37   \
stock_status                                                                 
instock           20    22    19    15    20     9    11    13     8     4   
outofstock         0     0     0     0     0     0     0     0     0     0   

stock_quantity   38    39    40    41    42    43    44    45    46    47   \
stock_status                                                                 
instock           11     7     7     4    11     6     3     6     1     2   
outofstock         0     0     0     0     0     0     0     0     0     0   

stock_quantity   48    49    50    51    52    53    54    55    58    62   \
stock_status                                                                 
instock            4     2     1     6     1     1     2     1     1     2   
outofstock         0     0     0     0     0     0     0     0     0     0   

stock_quantity   68    71    73    74    76    79    81    86    91    97   \
stock_status                                                                 
instock            1     1     1     1     1     1     2     1     1     1   
outofstock         0     0     0     0     0     0     0     0     0     0   

stock_quantity   98    100   101   103   112   115   121   123   125   127  \
stock_status                                                                 
instock            1     2     2     1     1     1     1     2     3     1   
outofstock         0     0     0     0     0     0     0     0     0     0   

stock_quantity   136   138   142   145  
stock_status                            
instock            1     1     1     2  
outofstock         0     0     0     0

In [ ]:
# Création d'une colonne "stock_status_2"
# La valeur de cette deuxième colonne sera fonction de la valeur dans la colonne "stock_quantity"
# Si la valeur de la colonne "stock_quantity" est nulle, renseigner "outofstock" sinon mettre "instock"
# Création de la colonne "stock_status_2"
df_erp["stock_status_2"] = df_erp["stock_quantity"].apply(
    lambda x: "outofstock" if pd.isna(x) or x == 0 else "instock"
)

# Vérification sur quelques lignes
df_erp[["stock_quantity", "stock_status_2"]].head(10)

stock_quantity stock_status_2
0              16        instock
1              10        instock
2               0     outofstock
3              26        instock
4               3        instock
5              12        instock
6              12        instock
7              15        instock
8               0     outofstock
9               5        instock

In [ ]:
# Vérifions que les 2 colonnes sont identiques:
# Les 2 colonnes sont strictement identiques si les valeurs de chaque ligne sont strictement identiques 2 à 2
# La comparaison de 2 colonnes peut se réaliser simplement avec l'instruction ci-dessous:
df_erp["stock_status"] == df_erp["stock_status_2"]

# Le résultat est l'affichage de True ou False pour chacune des lignes du dataset
# C'est un bon début, mais difficile à exploiter

0       True
1       True
2       True
3       True
4      False
       ...  
820     True
821     True
822     True
823     True
824     True
Length: 825, dtype: bool

In [ ]:
# Calcul de la somme des valeurs True (1) et False (0)
somme_identiques = (df_erp["stock_status"] == df_erp["stock_status_2"]).sum()

# Affichons le résultat
print(f"Nombre de lignes identiques : {somme_identiques}")
print(f"Nombre total de lignes : {len(df_erp)}")

# Vérifions si les deux colonnes sont strictement identiques
if somme_identiques == len(df_erp):
    print("✅ Les deux colonnes sont strictement identiques.")
else:
    print(f"⚠️ Il y a {len(df_erp) - somme_identiques} différence(s) entre les deux colonnes.")

Nombre de lignes identiques : 821
Nombre total de lignes : 825
⚠️ Il y a 4 différence(s) entre les deux colonnes.


<div style="
    background: linear-gradient(90deg, #ff7a00, #ff9500);
    color:black;
    padding:15px 40px;
    border-radius:10px;
    font-family:'Segoe UI', Arial, sans-serif;
    font-size:16px;
    text-align:center;
    box-shadow:0 4px 12px rgba(0,0,0,0.3);
    width:fit-content;
    margin:30px auto;
">
<b>Résultat :</b> 821/825 lignes identiques.<br>
⚠️ 4 différence(s) détectée(s).
</div>

In [ ]:
# Identifier les lignes où les valeurs diffèrent
differences = df_erp[df_erp["stock_status"] != df_erp["stock_status_2"]]

# Afficher les lignes en écart
display(differences)

product_id  onsale_web  price  stock_quantity stock_status  \
4          4039           1   46.0               3   outofstock   
398        4885           1   18.7               0      instock   
449        4973           0   10.0             -10   outofstock   
573        5700           1   44.5              -1   outofstock   

     purchase_price stock_status_2  
4             23.77        instock  
398            9.66     outofstock  
449            4.96        instock  
573           22.30        instock

In [ ]:
# Correction des incohérences entre stock_status, stock_status_2 et stock_quantity

# Logique de cohérence :
df_erp.loc[df_erp["stock_quantity"] > 0, ["stock_status", "stock_status_2"]] = "instock"
df_erp.loc[df_erp["stock_quantity"] <= 0, ["stock_status", "stock_status_2"]] = "outofstock"

# Vérification après correction
somme_identiques = (df_erp["stock_status"] == df_erp["stock_status_2"]).sum()
total_lignes = len(df_erp)
ecarts = total_lignes - somme_identiques

print(f"✅ Nombre de lignes identiques : {somme_identiques}")
print(f"📄 Nombre total de lignes : {total_lignes}")
if ecarts == 0:
    print("🟩 Les deux colonnes sont désormais cohérentes avec le stock.")
else:
    print(f"⚠️ {ecarts} différence(s) restante(s).")

✅ Nombre de lignes identiques : 825
📄 Nombre total de lignes : 825
🟩 Les deux colonnes sont désormais cohérentes avec le stock.


<div style="background:#4CAF50; color:black; padding:15px 40px; border-radius:10px;
font-family:Segoe UI; font-size:16px; text-align:center; width:fit-content; margin:30px auto;">
<b>Résultat :</b> 825/825 lignes identiques.<br>
✅ Les deux colonnes sont désormais cohérentes.
</div>

<a id="etape1_1"></a>
<div style="border: 1px solid RGB(0,150,250); border-radius:5px; background-color:rgba(51,165,182,0.05);">

  <h3 style="margin:auto; padding:15px 20px; color:RGB(51,165,182); display:flex; align-items:center;">
    <a href="#top" style="text-decoration:none; display:flex; align-items:center; margin-right:10px;">
      <img src="assets/img/house-solid-full.svg" width="16"
           style="vertical-align:middle; filter: invert(64%) sepia(53%) saturate(379%) hue-rotate(150deg) brightness(95%) contrast(92%);">
    </a>
    2.1.1 - Analyse exploratoire de chaque variable du fichier erp.xlsx
  </h3>
</div>

In [284]:
# Analyse exploratoire de chaque variable du fichier erp.xlsx

# Afficher les types de chaque colonne
print("📊 Types des variables :")
print(df_erp.dtypes)

# Vérifier les valeurs manquantes
print("\n❓ Valeurs manquantes :")
print(df_erp.isna().sum())

# Afficher le nombre de valeurs uniques par colonne
print("\n🔢 Nombre de valeurs uniques :")
print(df_erp.nunique())

# Afficher un résumé statistique pour les variables numériques
print("\n📈 Résumé statistique des variables numériques :")
display(df_erp.describe())

📊 Types des variables :
product_id          int64
onsale_web          int64
price             float64
stock_quantity      int64
stock_status       object
purchase_price    float64
stock_status_2     object
dtype: object

❓ Valeurs manquantes :
product_id        0
onsale_web        0
price             0
stock_quantity    0
stock_status      0
purchase_price    0
stock_status_2    0
dtype: int64

🔢 Nombre de valeurs uniques :
product_id        825
onsale_web          2
price             383
stock_quantity     84
stock_status        2
purchase_price    660
stock_status_2      2
dtype: int64

📈 Résumé statistique des variables numériques :


product_id  onsale_web       price  stock_quantity  purchase_price
count   825.000000  825.000000  825.000000      825.000000      825.000000
mean   5162.597576    0.867879   32.187697       21.589091       16.940582
std     902.644635    0.338828   26.712077       21.933807       14.561840
min    3847.000000    0.000000  -20.000000      -10.000000        2.740000
25%    4348.000000    1.000000   14.500000        7.000000        7.590000
50%    4907.000000    1.000000   24.300000       18.000000       12.710000
75%    5805.000000    1.000000   42.000000       30.000000       22.020000
max    7338.000000    1.000000  225.000000      145.000000      137.810000

<div style="font-family:Segoe UI, Arial; font-size:18px; color:#e0e0e0;
background-color:#1e1e1e; padding:20px; border-radius:10px; border:1px solid #444;
width:fit-content; margin:30px auto; text-align:left; box-shadow:0 0 15px #00000055;">

<p>🔍 <b>Analyse exploratoire du fichier <span style='color:#ffcc00;'>erp.xlsx</span></b></p>

<p>Le tableau comporte <span style='color:#ffcc00;font-weight:bold;'>825</span> observations
et <span style='color:#ffcc00;font-weight:bold;'>6</span> colonnes.</p>

<p>📦 Total de valeurs uniques (toutes colonnes confondues) :
<span style='color:#ffcc00;font-weight:bold;'>783</span></p>

<p>❗ Valeurs manquantes détectées :
<span style='color:lime;font-weight:bold;'>0</span></p>

</div>

<a id="etape2_1_1"></a>
<div style="border: 1px solid RGB(0,150,250); border-radius:5px; background-color:rgba(51,165,182,0.05);">

  <h3 style="margin:auto; padding:15px 20px; color:RGB(51,165,182); display:flex; align-items:center;">
    <a href="#top" style="text-decoration:none; display:flex; align-items:center; margin-right:10px;">
      <img src="assets/img/house-solid-full.svg" width="16"
           style="vertical-align:middle; filter: invert(64%) sepia(53%) saturate(379%) hue-rotate(150deg) brightness(95%) contrast(92%);">
    </a>
    2.1.1.1 - Analyse de la variable PRIX
  </h3>
</div>

In [285]:
###############
## LES PRIX  ##
###############

# Vérification des prix: Y a-t-il des prix non renseignés, négatifs ou nuls ?

# Afficher le ou les prix non renseignés dans la colonne "price"
print("Nombres d'articles avec un prix non renseigné: {}".format(df_erp["price"].isna().sum()))

# Afficher le prix minimum de la colonne "price"
print("Prix minimum :", df_erp["price"].min())

# Afficher le prix maximum de la colonne "price"
print("Prix maximum :", df_erp["price"].max())

# Afficher les prix inférieurs à 0 (qu'est-ce qu'il faut en faire ?)
print("Prix inférieurs à 0 :", df_erp[df_erp["price"] < 0])


Nombres d'articles avec un prix non renseigné: 0
Prix minimum : -20.0
Prix maximum : 225.0
Prix inférieurs à 0 :      product_id  onsale_web  price  stock_quantity stock_status  \
151        4233           0  -20.0               0   outofstock   
469        5017           0   -8.0               0   outofstock   
739        6594           0   -9.1              19      instock   

     purchase_price stock_status_2  
151           10.33     outofstock  
469            4.34     outofstock  
739            4.61        instock  


<div style="font-family:Segoe UI, Arial; font-size:17px; color:#e0e0e0;
background-color:#1e1e1e; padding:20px; border-radius:10px; border:1px solid #444;
width:fit-content; margin:30px auto; text-align:left; box-shadow:0 0 15px #00000055;">

<p>💰 <b>Analyse de la variable <span style='color:#ffcc00;'>price</span></b></p>

<p>📊 Aucun prix manquant détecté :
<span style='color:lime; font-weight:bold;'>0</span></p>

<p>🔽 <b>Prix minimum :</b> <span style='color:#ff6a00;'>-20.0</span><br>
🔼 <b>Prix maximum :</b> <span style='color:#ffcc00;'>225.0</span></p>

<p>⚠️ <b>Produits avec un prix négatif :</b> <span style='color:#ff8080; font-weight:bold;'>3</span><br>
<small>(Les valeurs négatives devront être corrigées ou remplacées par 0.)</small></p>

</div>

In [286]:
# Correction des incohérences détectées dans la variable "price"

# Remplacer les NaN (au cas où certains apparaîtraient encore)
df_erp["price"] = df_erp["price"].fillna(0)

# Corriger les prix négatifs
prix_negatifs = (df_erp["price"] < 0).sum()
df_erp.loc[df_erp["price"] < 0, "price"] = 0

# Vérifier après correction
prix_min = df_erp["price"].min()
prix_max = df_erp["price"].max()
prix_manquants = df_erp["price"].isna().sum()

print(f"✅ Prix corrigés avec succès.\nPrix min : {prix_min}\nPrix max : {prix_max}\nPrix manquants : {prix_manquants}")

✅ Prix corrigés avec succès.
Prix min : 0.0
Prix max : 225.0
Prix manquants : 0


<div style="background:#4CAF50; color:black; padding:15px 40px; border-radius:10px;
font-family:Segoe UI; font-size:16px; text-align:center; width:fit-content; margin:30px auto;">
<b>Résultat :</b><br>
✅ Toutes les incohérences sur la variable <b>price</b> ont été corrigées.<br>
Les valeurs manquantes ont été remplacées et les prix négatifs ajustés à <b>0</b>.<br>
📊 Les données sont désormais prêtes pour l’analyse.
</div>

<a id="etape2_1_1_2"></a>
<div style="border: 1px solid RGB(0,150,250); border-radius:5px; background-color:rgba(51,165,182,0.05);">

  <h3 style="margin:auto; padding:15px 20px; color:RGB(51,165,182); display:flex; align-items:center;">
    <a href="#top" style="text-decoration:none; display:flex; align-items:center; margin-right:10px;">
      <img src="assets/img/house-solid-full.svg" width="16"
           style="vertical-align:middle; filter: invert(64%) sepia(53%) saturate(379%) hue-rotate(150deg) brightness(95%) contrast(92%);">
    </a>
    2.1.1.2 - Analyse de la variable STOCK
  </h3>
</div>

In [287]:
#####################
### stock_quantity ###
#####################

# Vérification de la colonne stock_quantity
# Afficher la quantité minimum de la colonne "stock_quantity"
print("Quantité minimum :", df_erp["stock_quantity"].min())

# Afficher la quantité maximum de la colonne "stock_quantity"
print("Quantité maximum :", df_erp["stock_quantity"].max())

# Afficher les stocks inférieurs à 0 (qu'est-ce qu'il faut en faire ?)
stocks_negatifs = df_erp[df_erp["stock_quantity"] < 0]
print("Stocks inférieurs à 0 :")
display(stocks_negatifs)

Quantité minimum : -10
Quantité maximum : 145
Stocks inférieurs à 0 :


product_id  onsale_web  price  stock_quantity stock_status  \
449        4973           0   10.0             -10   outofstock   
573        5700           1   44.5              -1   outofstock   

     purchase_price stock_status_2  
449            4.96     outofstock  
573           22.30     outofstock

<div style="font-family:Segoe UI, Arial; font-size:17px; color:#e0e0e0;
background-color:#1e1e1e; padding:20px; border-radius:10px; border:1px solid #444;
width:fit-content; margin:30px auto; text-align:left; box-shadow:0 0 15px #00000055;">

<p>📦 <b>Analyse de la variable <span style='color:#ffcc00;'>stock_quantity</span></b></p>

<p>🔽 <b>Quantité minimum :</b> <span style='color:#ff6a00;'>-10</span><br>
🔼 <b>Quantité maximum :</b> <span style='color:#ffcc00;'>145</span></p>

<p>⚠️ <b>Produits avec un stock négatif :</b> <span style='color:#ff8080; font-weight:bold;'>3</span><br>
<small>(Les valeurs négatives indiquent une erreur de saisie et doivent être corrigées.)</small></p>

</div>

In [288]:
# Correction des incohérences dans la variable "stock_quantity"

# Identifier combien de valeurs négatives il y avait
stocks_negatifs = (df_erp["stock_quantity"] < 0).sum()
print(f"⚠️ Nombre de produits avec stock négatif avant correction : {stocks_negatifs}")

# Corriger les valeurs négatives en les remplaçant par 0
df_erp.loc[df_erp["stock_quantity"] < 0, "stock_quantity"] = 0

# Vérifier après correction
stock_min = df_erp["stock_quantity"].min()
stock_max = df_erp["stock_quantity"].max()
stocks_negatifs_apres = (df_erp["stock_quantity"] < 0).sum()

print("\n✅ Correction effectuée avec succès.")
print(f"Quantité minimale après correction : {stock_min}")
print(f"Quantité maximale : {stock_max}")
print(f"Stocks négatifs restants : {stocks_negatifs_apres}")

⚠️ Nombre de produits avec stock négatif avant correction : 2

✅ Correction effectuée avec succès.
Quantité minimale après correction : 0
Quantité maximale : 145
Stocks négatifs restants : 0


<div style="background:#4CAF50; color:black; padding:15px 40px; border-radius:10px;
font-family:Segoe UI; font-size:16px; text-align:center; width:fit-content; margin:30px auto;">
<b>Résultat :</b><br>
✅ Toutes les valeurs négatives de <b>stock_quantity</b> ont été remplacées par <b>0</b>.<br>
📦 Les stocks sont désormais cohérents et prêts pour les opérations de fusion ou d’analyse.
</div>

<a id="etape2_1_1_3"></a>
<div style="border: 1px solid RGB(0,150,250); border-radius:5px; background-color:rgba(51,165,182,0.05);">

  <h3 style="margin:auto; padding:15px 20px; color:RGB(51,165,182); display:flex; align-items:center;">
    <a href="#top" style="text-decoration:none; display:flex; align-items:center; margin-right:10px;">
      <img src="assets/img/house-solid-full.svg" width="16"
           style="vertical-align:middle; filter: invert(64%) sepia(53%) saturate(379%) hue-rotate(150deg) brightness(95%) contrast(92%);">
    </a>
    2.1.1.3 - Analyse de la variable ONSALE_WEB
  </h3>
</div>

In [289]:
#Vérification de la colonne onsale_web et des valeurs qu'elle contient. Que signifient-elles?
print("Valeurs uniques de la colonne 'onsale_web' :", df_erp["onsale_web"].unique())
print("\nFréquence de chaque valeur :\n", df_erp["onsale_web"].value_counts())

# Interprétation :
# 0 signifie que le produit n'est pas en vente sur le site web
# 1 signifie que le produit est en vente en ligne

Valeurs uniques de la colonne 'onsale_web' : [1 0]

Fréquence de chaque valeur :
 onsale_web
1    716
0    109
Name: count, dtype: int64


In [290]:
#Quelles sont les colonnes à conserver selon vous?
# Colonnes pertinentes à conserver pour l’analyse
colonnes_a_conserver = ["product_id", "onsale_web", "price", "purchase_price", "stock_quantity", "stock_status", "stock_status_2"]
df_erp = df_erp[colonnes_a_conserver]

print("Colonnes conservées :", list(df_erp.columns))

Colonnes conservées : ['product_id', 'onsale_web', 'price', 'purchase_price', 'stock_quantity', 'stock_status', 'stock_status_2']


In [291]:
#Supprimer la colonne comportant le libellé "stock_status_2" car elle est redondante 
#avec la colonne "stock_status".
# Suppression de la colonne redondante
df_erp = df_erp.drop(columns=["stock_status_2"], errors="ignore")

print("Colonnes restantes après suppression :", list(df_erp.columns))

Colonnes restantes après suppression : ['product_id', 'onsale_web', 'price', 'purchase_price', 'stock_quantity', 'stock_status']


<div style="font-family:Segoe UI, Arial; font-size:17px; color:#e0e0e0;
background-color:#1e1e1e; padding:20px; border-radius:10px; border:1px solid #444;
width:fit-content; margin:30px auto; text-align:left; box-shadow:0 0 15px #00000055;">

<p>🌐 <b>Analyse de la variable <span style='color:#ffcc00;'>onsale_web</span></b></p>

<p>Cette variable indique si le produit est disponible à la vente sur le site web :<br>
<span style='color:#ffcc00;'>1</span> ➜ En ligne<br>
<span style='color:#ffcc00;'>0</span> ➜ Non disponible</p>

<p>📊 La majorité des produits sont disponibles en ligne (716 contre 109 hors ligne).</p>

<p>✅ Colonnes conservées : product_id, onsale_web, price, purchase_price, stock_quantity, stock_status</p>
<p>🧹 Colonne <span style='color:#ff6a00;'>stock_status_2</span> supprimée car redondante.</p>

</div>

<a id="etape2_1_1_4"></a>
<div style="border: 1px solid RGB(0,150,250); border-radius:5px; background-color:rgba(51,165,182,0.05);">

  <h3 style="margin:auto; padding:15px 20px; color:RGB(51,165,182); display:flex; align-items:center;">
    <a href="#top" style="text-decoration:none; display:flex; align-items:center; margin-right:10px;">
      <img src="assets/img/house-solid-full.svg" width="16"
           style="vertical-align:middle; filter: invert(64%) sepia(53%) saturate(379%) hue-rotate(150deg) brightness(95%) contrast(92%);">
    </a>
    2.1.1.4 - Analyse de la variable prix d'achat
  </h3>
</div>

In [292]:
######################
##   prix d'achat   ##
######################
# Vérification de la colonne purchase_price
print("Nombres d'articles avec un prix d'achat non renseigné :", df_erp["purchase_price"].isna().sum())

# Afficher le prix minimum de la colonne "purchase_price"
print("Prix d'achat minimum :", df_erp["purchase_price"].min())

# Afficher le prix maximum de la colonne "purchase_price"
print("Prix d'achat maximum :", df_erp["purchase_price"].max())

Nombres d'articles avec un prix d'achat non renseigné : 0
Prix d'achat minimum : 2.74
Prix d'achat maximum : 137.81


<div style="font-family:Segoe UI, Arial; font-size:17px; color:#e0e0e0;
background-color:#1e1e1e; padding:20px; border-radius:10px; border:1px solid #444;
width:fit-content; margin:30px auto; text-align:left; box-shadow:0 0 15px #00000055;">

<p>💰 <b>Analyse de la variable <span style='color:#ffcc00;'>purchase_price</span></b></p>

<p>🧾 Nombre d’articles avec un prix d’achat non renseigné : 
<span style='color:lime; font-weight:bold;'>0</span></p>

<p>📉 Prix d’achat minimum : 
<span style='color:#ff6a00; font-weight:bold;'>2.74</span><br>
📈 Prix d’achat maximum : 
<span style='color:#ffcc00; font-weight:bold;'>137.81</span></p>

</div>

<a id="etape2_2"></a>
<div style="border: 1px solid RGB(0,150,250); border-radius:5px; background-color:rgba(51,165,182,0.05);">

  <h3 style="margin:auto; padding:15px 20px; color:RGB(51,165,182); display:flex; align-items:center;">
    <a href="#top" style="text-decoration:none; display:flex; align-items:center; margin-right:10px;">
      <img src="assets/img/house-solid-full.svg" width="16"
           style="vertical-align:middle; filter: invert(64%) sepia(53%) saturate(379%) hue-rotate(150deg) brightness(95%) contrast(92%);">
    </a>
    2.2 - Analyse exploratoire du fichier web.xlsx
  </h3>
</div>

In [293]:
#Dimension du dataset
#Nombre d'observations

#Nombre de caractéristiques


In [294]:
#Consulter le nombre de colonnes
#La nature des données dans chacune des colonnes
#Le nombre de valeurs présentes dans chacune des colonnes


In [295]:
#Selon vous, quelles sont les colonnes à conserver ?


In [296]:
#Si vous avez défini des colonnes à supprimer, effectuer l'opération


In [297]:
#Visualisation des valeurs de la colonne sku
#Quelles sont les valeurs qui ne semblent pas respecter la régle de codification?


In [298]:
#Si vous avez identifié des codes articles ne respectant pas la régle de codification, consultez-les


In [299]:
#Identifier les lignes sans code article


In [300]:
#Pour les codes articles identifiés, réaliser une analyse et définir l'action à entreprendre


In [301]:
#La clé pour chaque ligne est-elle unique? autrement dit, y a-t-il des doublons?


In [302]:
#Les lignes sans code article semblent être toutes non renseignées
#Pour s'en assurer, réaliser les étapes suivantes:
#1 - Créer un dataframe avec uniquement les lignes sans code article

#2 - Utiliser la fonction df.info() sur ce nouveau dataframe pour observer le nombre de valeurs renseignées dans chacune des colonnes

#3 - Que constatez-vous?

<a id="etape2_3"></a>
<div style="border: 1px solid RGB(0,150,250); border-radius:5px; background-color:rgba(51,165,182,0.05);">

  <h3 style="margin:auto; padding:15px 20px; color:RGB(51,165,182); display:flex; align-items:center;">
    <a href="#top" style="text-decoration:none; display:flex; align-items:center; margin-right:10px;">
      <img src="assets/img/house-solid-full.svg" width="16"
           style="vertical-align:middle; filter: invert(64%) sepia(53%) saturate(379%) hue-rotate(150deg) brightness(95%) contrast(92%);">
    </a>
    2.3 - Analyse exploratoire du fichier liaison.xlsx
  </h3>
</div>

In [303]:
#Dimension du dataset
#Nombre d'observations

#Nombre de caractéristiques


In [304]:
#Consulter le nombre de colonnes
#La nature des données dans chacune des colonnes
#Le nombre de valeurs présentes dans chacune des colonnes


In [305]:
#Les valeurs de la colonne "product_id" sont-elles toutes uniques?


In [306]:
#Les valeurs de la colonne "id_web" sont-elles toutes uniques?


In [307]:
#Avons-nous des articles sans correspondance?


<a id="etape3"></a>
<div style="
  border: 1px solid RGB(0,150,250);
  border-radius:5px;
  background-color:rgba(125, 194, 204, 0.15);
  margin:auto;
">

  <h2 style="
    margin:auto;
    padding:15px 20px;
    background-color:RGB(51,165,182);
    color:#fff;
    display:flex;
    align-items:center;
    border-radius:5px;
  ">
    <a href="#top" style="text-decoration:none; display:flex; align-items:center; margin-right:10px;">
      <img src="assets/img/house-solid-full.svg" width="30" style="vertical-align:middle;">
    </a>
    Étape 3 - Jonction des fichiers
  </h2>
</div>

<a id="etape3_1"></a>
<div style="border: 1px solid RGB(0,150,250); border-radius:5px; background-color:rgba(51,165,182,0.05);">

  <h3 style="margin:auto; padding:15px 20px; color:RGB(51,165,182); display:flex; align-items:center;">
    <a href="#top" style="text-decoration:none; display:flex; align-items:center; margin-right:10px;">
      <img src="assets/img/house-solid-full.svg" width="16"
           style="vertical-align:middle; filter: invert(64%) sepia(53%) saturate(379%) hue-rotate(150deg) brightness(95%) contrast(92%);">
    </a>
    Etape 3.1 - Jonction du fichier df_erp et df_liaison
  </h3>
</div>

In [308]:
#Fusion des fichiers df_erp et df_liaison


In [309]:
#Y a t-il des lignes ne "matchant" pas entre les 2 fichiers?


<a id="etape3_2"></a>
<div style="border: 1px solid RGB(0,150,250); border-radius:5px; background-color:rgba(51,165,182,0.05);">

  <h3 style="margin:auto; padding:15px 20px; color:RGB(51,165,182); display:flex; align-items:center;">
    <a href="#top" style="text-decoration:none; display:flex; align-items:center; margin-right:10px;">
      <img src="assets/img/house-solid-full.svg" width="16"
           style="vertical-align:middle; filter: invert(64%) sepia(53%) saturate(379%) hue-rotate(150deg) brightness(95%) contrast(92%);">
    </a>
    Etape 3.2 - Jonction du fichier df_merge et df_web
  </h3>
</div>

In [310]:
#Fusionner les datasets df_merge et df_web


In [311]:
#Avons-nous des lignes sans correspondance?


<a id="etape4"></a>
<div style="
  border: 1px solid RGB(0,150,250);
  border-radius:5px;
  background-color:rgba(125, 194, 204, 0.15);
  margin:auto;
">

  <h2 style="
    margin:auto;
    padding:15px 20px;
    background-color:RGB(51,165,182);
    color:#fff;
    display:flex;
    align-items:center;
    border-radius:5px;
  ">
    <a href="#top" style="text-decoration:none; display:flex; align-items:center; margin-right:10px;">
      <img src="assets/img/house-solid-full.svg" width="30" style="vertical-align:middle;">
    </a>
    Etape 4 - Analyse univariée des prix
  </h2>
</div>

<a id="etape4_1"></a>
<div style="border: 1px solid RGB(0,150,250); border-radius:5px; background-color:rgba(51,165,182,0.05);">

  <h3 style="margin:auto; padding:15px 20px; color:RGB(51,165,182); display:flex; align-items:center;">
    <a href="#top" style="text-decoration:none; display:flex; align-items:center; margin-right:10px;">
      <img src="assets/img/house-solid-full.svg" width="16"
           style="vertical-align:middle; filter: invert(64%) sepia(53%) saturate(379%) hue-rotate(150deg) brightness(95%) contrast(92%);">
    </a>
    Etape 4.1 - Exploration par la visualisation de données
  </h3>
</div>

In [312]:
#Création d'une boîte à moustache de la répartition des prix grâce à Pandas


In [313]:
#Autre méthode avec plotly express


<a id="etape4_2"></a>
<div style="border: 1px solid RGB(0,150,250); border-radius:5px; background-color:rgba(51,165,182,0.05);">

  <h3 style="margin:auto; padding:15px 20px; color:RGB(51,165,182); display:flex; align-items:center;">
    <a href="#top" style="text-decoration:none; display:flex; align-items:center; margin-right:10px;">
      <img src="assets/img/house-solid-full.svg" width="16"
           style="vertical-align:middle; filter: invert(64%) sepia(53%) saturate(379%) hue-rotate(150deg) brightness(95%) contrast(92%);">
    </a>
    Etape 4.2 - Exploration par l'utilisation de méthodes statistiques
  </h3>
</div>

<a id="etape4_2_1"></a>
<div style="border: 1px solid RGB(0,150,250); border-radius:5px; background-color:rgba(51,165,182,0.05);">

  <h3 style="margin:auto; padding:15px 20px; color:RGB(51,165,182); display:flex; align-items:center;">
    <a href="#top" style="text-decoration:none; display:flex; align-items:center; margin-right:10px;">
      <img src="assets/img/house-solid-full.svg" width="16"
           style="vertical-align:middle; filter: invert(64%) sepia(53%) saturate(379%) hue-rotate(150deg) brightness(95%) contrast(92%);">
    </a>
    Etape 4.2.1 - Identification par le Z-index
  </h3>
</div>

In [314]:
#Calculer la moyenne du prix

#Calculer l'écart-type du prix

#Calculer le Z-score


In [315]:
#Quel est le seuil prix dont le z-score est supérieur à 3?


<a id="etape4_2_2"></a>
<div style="border: 1px solid RGB(0,150,250); border-radius:5px; background-color:rgba(51,165,182,0.05);">

  <h3 style="margin:auto; padding:15px 20px; color:RGB(51,165,182); display:flex; align-items:center;">
    <a href="#top" style="text-decoration:none; display:flex; align-items:center; margin-right:10px;">
      <img src="assets/img/house-solid-full.svg" width="16"
           style="vertical-align:middle; filter: invert(64%) sepia(53%) saturate(379%) hue-rotate(150deg) brightness(95%) contrast(92%);">
    </a>
    Etape 4.2.2 - Identification par l'intervalle interquartile
  </h3>
</div>

In [316]:
#Utilisation de la fonction "describe" de Pandas pour l'étude des mesures de dispersion


In [317]:
#Définir un seuil pour les articles "outliers" en prix


In [318]:
#Définir le nombre d'articles et la proportion de l'ensemble du catalogue "outliers"


In [319]:
#Selon vous, ces outliers sont-ils justifiés ? Comment le démontrer si cela est possible ?


<a id="etape5"></a>
<div style="
  border: 1px solid RGB(0,150,250);
  border-radius:5px;
  background-color:rgba(125, 194, 204, 0.15);
  margin:auto;
">

  <h2 style="
    margin:auto;
    padding:15px 20px;
    background-color:RGB(51,165,182);
    color:#fff;
    display:flex;
    align-items:center;
    border-radius:5px;
  ">
    <a href="#top" style="text-decoration:none; display:flex; align-items:center; margin-right:10px;">
      <img src="assets/img/house-solid-full.svg" width="40" style="vertical-align:middle;">
    </a>
    Etape 5 - Analyse univariée du CA, des quantités vendues, des stocks et de la marge ainsi qu'une analyse multivariée
  </h2>
</div>

<a id="etape5_1"></a>
<div style="border: 1px solid RGB(0,150,250); border-radius:5px; background-color:rgba(51,165,182,0.05);">

  <h3 style="margin:auto; padding:15px 20px; color:RGB(51,165,182); display:flex; align-items:center;">
    <a href="#top" style="text-decoration:none; display:flex; align-items:center; margin-right:10px;">
      <img src="assets/img/house-solid-full.svg" width="16"
           style="vertical-align:middle; filter: invert(64%) sepia(53%) saturate(379%) hue-rotate(150deg) brightness(95%) contrast(92%);">
    </a>
    Etape 5.1 - Analyse des ventes en CA
  </h3>
</div>

In [320]:
##############################
# Calculer le CA du site web #
##############################

#Créer une colonne calculant le CA par article

#Calculer la somme de la colonne "ca_par_article"
#Ce résultat correspond au chiffre d'affaire du site web


In [321]:
###############################
# Palmarès des articles en CA #
###############################

#Effectuer le tri dans l'ordre décroissant du CA du dataset df_merge

#Réinitialiser l'index du dataset par un reset_index

#Afficher les 20 premiers articles en CA

#Graphique en barre des 20 premiers articles avec plotly express


In [322]:
#############################
# Calculer le 20 / 80 en CA #
#############################

#Créer une colonne calculant la part du CA de la ligne dans le dataset

#Créer une colonne réalisant la somme cumulative de la colonne précedemment créée

#Grâce aux deux colonnes créées précedemment, calculer le nombre d'articles représentant 80% du CA

#Afficher la proportion que représente ce groupe d'articles dans le catalogue entier du site web


<a id="etape5_2"></a>
<div style="border: 1px solid RGB(0,150,250); border-radius:5px; background-color:rgba(51,165,182,0.05);">

  <h3 style="margin:auto; padding:15px 20px; color:RGB(51,165,182); display:flex; align-items:center;">
    <a href="#top" style="text-decoration:none; display:flex; align-items:center; margin-right:10px;">
      <img src="assets/img/house-solid-full.svg" width="16"
           style="vertical-align:middle; filter: invert(64%) sepia(53%) saturate(379%) hue-rotate(150deg) brightness(95%) contrast(92%);">
    </a>
    Etape 5.2 - Analyse des ventes en quantité
  </h3>
</div>

In [323]:
#####################################
# Palmarès des articles en quantité #
#####################################

#Effectuer le tri dans l'ordre décroissant de quantités vendues du dataset df_merge

#Réinitialiser l'index du dataset par un reset_index

#Afficher les 20 premiers articles en quantité

#Graphique en barre des 20 premiers articles avec plotly express


In [324]:
#############################
# Calculer le 20 / 80 en CA #
#############################

#Créer une colonne calculant la part en quantité de la ligne dans le dataset

#Créer une colonne réalisant la somme cumulative de la colonne précedemment créée

#Grâce aux deux colonnes créées précedemment, calculer le nombre d'articles représentant 80% des ventes en quantité

#Afficher la proportion que représente ce groupe d'articles dans le catalogue entier du site web


<a id="etape5_3"></a>
<div style="border: 1px solid RGB(0,150,250); border-radius:5px; background-color:rgba(51,165,182,0.05);">

  <h3 style="margin:auto; padding:15px 20px; color:RGB(51,165,182); display:flex; align-items:center;">
    <a href="#top" style="text-decoration:none; display:flex; align-items:center; margin-right:10px;">
      <img src="assets/img/house-solid-full.svg" width="16"
           style="vertical-align:middle; filter: invert(64%) sepia(53%) saturate(379%) hue-rotate(150deg) brightness(95%) contrast(92%);">
    </a>
    Etape 5.3 - Analyse des stocks
  </h3>
</div>

In [325]:
######################################
# Calculer le nombre de mois de stock #
######################################

#Import de numpy 

#Création de la colonne Rotation de stock

#Remplacement des "inf" par 0

#Effectuer le tri dans l'ordre décroissant du nombre de mois de stock dans le dataset df_merge

#Graphique en barre du flop 20 des produits qui ont le plus de mois de stock

In [326]:
####################################
# Valorisation des stocks en euros #
####################################

#Création de la colonne Valorisation des stocks en euros

#Calculer la somme de la colonne "Valorisation_stock_euros"

In [327]:
##############################################
# Valorisation du nombre de produits en stock #
##############################################

#Calculer la somme de la colonne stock quantity

<a id="etape5_4"></a>
<div style="border: 1px solid RGB(0,150,250); border-radius:5px; background-color:rgba(51,165,182,0.05);">

  <h3 style="margin:auto; padding:15px 20px; color:RGB(51,165,182); display:flex; align-items:center;">
    <a href="#top" style="text-decoration:none; display:flex; align-items:center; margin-right:10px;">
      <img src="assets/img/house-solid-full.svg" width="16"
           style="vertical-align:middle; filter: invert(64%) sepia(53%) saturate(379%) hue-rotate(150deg) brightness(95%) contrast(92%);">
    </a>
    Etape 5.4 - Analyse du taux de marge
  </h3>
</div>

In [328]:
############################
# Analyse du taux de marge #
############################

#Création de la colonne Prix HT

#Création de la colonne Taux de marge

#Afficher le prix minimum de la colonne "taux_marge"

#Afficher le prix maximum de la colonne "taux_marge"


In [329]:
#Affichage de la ligne avec un taux de marge inférieur à 0


In [330]:
#Création d'un dataframe avec les taux positifs

#Afficher le prix minimum de la colonne "taux_marge"

#Afficher le prix maximum de la colonne "taux_marge"


In [331]:
#Création d'un dataframe avec le taux de marge moyen par type de produit

#Affichage dans un graphique du taux de marge par type de produit


<a id="etape5_5"></a>
<div style="border: 1px solid RGB(0,150,250); border-radius:5px; background-color:rgba(51,165,182,0.05);">

  <h3 style="margin:auto; padding:15px 20px; color:RGB(51,165,182); display:flex; align-items:center;">
    <a href="#top" style="text-decoration:none; display:flex; align-items:center; margin-right:10px;">
      <img src="assets/img/house-solid-full.svg" width="16"
           style="vertical-align:middle; filter: invert(64%) sepia(53%) saturate(379%) hue-rotate(150deg) brightness(95%) contrast(92%);">
    </a>
    Etape 5.5 - Analyse des corrélations entre les variables stock, sales et price
  </h3>
</div>

In [332]:
############################
# Analyse des corrélations #
############################

#Importation de Seaborn

#Création d'une heatmap de corrélation avec les variables stock, sales et price
#On peut également créer un mask pour n'afficher qu'une demi heatmap

In [333]:
#Que peut-on conclure des corrélations ?

<a id="etape5_6"></a>
<div style="border: 1px solid RGB(0,150,250); border-radius:5px; background-color:rgba(51,165,182,0.05);">

  <h3 style="margin:auto; padding:15px 20px; color:RGB(51,165,182); display:flex; align-items:center;">
    <a href="#top" style="text-decoration:none; display:flex; align-items:center; margin-right:10px;">
      <img src="assets/img/house-solid-full.svg" width="16"
           style="vertical-align:middle; filter: invert(64%) sepia(53%) saturate(379%) hue-rotate(150deg) brightness(95%) contrast(92%);">
    </a>
    Etape 5.6 - Mise à disposition de la nouvelle table sur un fichier Excel
  </h3>
</div>

In [334]:
#Mettre le dataset df_merge sur un fichier Excel
#Cette étape peut être utile pour partager le résultat du dataset obtenu avec les équipes.  
